contracts/TrueCurrencies/TrueRewardBackedToken.sol

```diff
@@ -230,153 +229,95 @@ contract TrueRewardBackedToken is RewardTokenWithReserve {
         return _toToken(opportunityRewardSupply(), opportunity());
     }
 
-    /**
-     * @dev Transfer helper for accounts with rewardToken balances
-     * Uses reserve float to save gas costs for transactions with value < reserve balance
-     * Case #2 and #3 use reserve balances
-     *
-     * There are 6 transfer cases
-     *  1. Both sender and receiver are disabled (see _transferAllArgs)
-     *  2. Sender enabled, receiver disabled, value < reserve TrueCurrency balance
-     *  3. Sender disabled, receiver enabled, value < reserve rewardToken balance (in TrueCurrency)
-     *  4. Both sender and receiver are enabled
-     *  5. Sender enabled, receiver disabled, value > reserve TrueCurrency balance
-     *  6. Sender disabled, receiver enabled, value > reserve rewardToken balance (in TrueCurrency)
-     *
-     * @param _from account to transfer from
-     * @param _to account to transfer to
-     * @param _value value in Token to transfer
-     * @return actual value transferred
-     */
-    function _transferWithRewards(
-        address _from,
-        address _to,
-        uint256 _value
-    ) internal returns (uint256) {
-        // get enable stat
-        bool fromEnabled = trueRewardEnabled(_from);
-        bool toEnabled = trueRewardEnabled(_to);
-
-        // get opportunity address
-        address finOp = opportunity();
-
-        // calculate rewardToken balance
-        uint rewardAmount = _toRewardToken(_value, finOp);
-
-        // 2. Sender enabled, receiver disabled, value < reserve TrueCurrency balance
-        // Swap rewardToken for Token through reserve
-        if (fromEnabled && !toEnabled && _value <= reserveBalance()) {
-            swapRewardForToken(_from, _to, _value, finOp);
-        }
-        // 3. Sender disabled, receiver enabled, value < reserve rewardToken balance
-        // Swap Token for rewardToken through reserve
-        else if (!fromEnabled && toEnabled && rewardAmount <= rewardTokenBalance(RESERVE, finOp)) {
-            swapTokenForReward(_from, _to, _value, finOp);
+    function redeemWithReserve(address sender, uint256 depositAmount, uint256 rewardAmount, address finOp) internal returns (uint256) {
+        if (reserveBalance() >= depositAmount) {
+            swapRewardForToken(sender, depositAmount, rewardAmount, finOp);
+            return depositAmount;
+        } else {
+            return redeemRewardToken(sender, rewardAmount, finOp);
         }
-        // 4. Sender and receiver are enabled
-        // Here we simply transfer rewardToken from the sender to the receiver
-        else if (fromEnabled && toEnabled) {
-            _subRewardBalance(_from, rewardAmount, finOp);
-            _addRewardBalance(_to, rewardAmount, finOp);
+    }
+
+    function redeemFromSender(address sender, uint256 depositAmount) internal returns (uint256) {
```

Comment:
remove this function, just check in transfer function and call redeemWithReserve